src/ui/include/EditorNS/editor.h

```diff
@@ -227,7 +222,10 @@ namespace EditorNS
         Q_INVOKABLE void setZoomFactor(const qreal &factor);
         Q_INVOKABLE void setSelectionsText(const QStringList &texts, selectMode mode);
         Q_INVOKABLE void setSelectionsText(const QStringList &texts);
-        Q_INVOKABLE QString language();
+        Q_INVOKABLE QString getLanguageId() {return ((m_currentLanguage != nullptr) ? m_currentLanguage->id : "");}
+        Q_INVOKABLE QString getLanguageName() {return ((m_currentLanguage != nullptr) ? m_currentLanguage->name : "");}
+        Q_INVOKABLE QString getLanguageMime() {return ((m_currentLanguage != nullptr) ? m_currentLanguage->mime : "");}
+        Q_INVOKABLE QString getLanguageMode() {return ((m_currentLanguage != nullptr) ? m_currentLanguage->mode : "");}
```

Comment:
In order to avoid all the nullptr checks perhaps we should just set m_currentLanguage to the plaintext language initially and ensure that calls to setLanguage() always either assign the correct new language to m_currentLanguage or plaintext.

Logically we'd never want an Editor that has no assigned language, so we could treat m_currentLanguage as a "non-null ptr" by ensuring we never set to it nullptr.